### ISSUE TITLE
Error when I import import 'package:network_to_file_image/network_to_file_image.dart';

### ISSUE THREAD

**Issue by jodymac** [marcglasberg/network_to_file_image#14]
IOS build. I've tried beta, stable and master.  I've downgraded flutter in multiple channels, I've removed all the versions cached of network_to_file image from my flutter directory.  I've used this package in other apps, they won't compile either now.

Xcode and flutter did just update in the last two days.

Compiler message:
../../../flutter/flutter/.pub-cache/hosted/pub.dartlang.org/network_to_file_imag
e-2.2.8/lib/network_to_file_image.dart:732:8: Error: The method
'_MockHttpHeaders.add' has fewer named arguments than those of overridden method
'HttpHeaders.add'.
  void add(String name, Object value) {}                                
       ^                                                                
org-dartlang-sdk:///third_party/dart/sdk/lib/_http/http.dart:695:8: Context:
This is the overridden method ('add').
  void add(String name, Object value,                                   
       ^                                                                
../../../flutter/flutter/.pub-cache/hosted/pub.dartlang.org/network_to_file_imag
e-2.2.8/lib/network_to_file_image.dart:750:8: Error: The method
'_MockHttpHeaders.set' has fewer named arguments than those of overridden method
'HttpHeaders.set'.
  void set(String name, Object value) {}                                
       ^                                                                
org-dartlang-sdk:///third_party/dart/sdk/lib/_http/http.dart:704:8: Context:
This is the overridden method ('set').
  void set(String name, Object value,   

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.18.0-6.0.pre.83, on Mac OS X 10.15.4 19E287,
    locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.44.2)
[✓] Connected device (4 available)

**Comment by marcglasberg** [comment]
This is the class it's complaining about:

```dart
class _MockHttpHeaders extends HttpHeaders {
  //
  @override
  List<String> operator [](String name) => <String>[];

  @override
  void add(String name, Object value) {}
   ...
```

And this is the signature of the overriden method from : `http.dart`

```dart
  /**
   * Adds a header value. The header named [name] will have the value
   * [value] added to its list of values. Some headers are single
   * valued, and for these adding a value will replace the previous
   * value. If the value is of type DateTime a HTTP date format will be
   * applied. If the value is a [:List:] each element of the list will
   * be added separately. For all other types the default [:toString:]
   * method will be used.
   */
  void add(String name, Object value);
```

They match, as you can see.

Could you please post here the copy of your corresponding `add()` method from your `http.dart`?

**Comment by jodymac** [comment]
Thank you for the very quick response!

Here is what I have …
  void add(String name, Object value,
      {@Since("2.8") bool preserveHeaderCase = false});

This was located at flutter/bin/cache/dart-sdk/lib/_http/http.dart on line 695

If this is not where I am to be looking, please tell me where to look.

**Comment by marcglasberg** [comment]
Ok, problem is you are in master now, and clearly if you go to the stable channel you'll get the current signature and it will work.

You said you downgraded Flutter in multiple channels, but you probably did something wrong. Try doing `flutter channel stable` and then `flutter upgrade`.

This signature is only adding an optional parameter, so I may be able to fix this for the master channel without leaving the stable channel, if and only if all methods in that file only added optional parameters:

This:

    @override
    void add(String name, Object value, {bool preserveHeaderCase = false}) 

Can override this:

    void add(String name, Object value) 

However, even if I fix the signatures, it's not a guarantee that it will work in master, because maybe they did some other changes.

**Comment by jodymac** [comment]
Last night I went from Beta (working on web app), to Stable, to Master.  I did do flutter upgrade after each change.  Even went back to stable and used the downgrade command I had never used before to bring my flutter version down to previous versions that had worked.

I agree, somewhere I did something wrong, because moving back to stable today didn’t give me an error on network_to_file_image when I ran the app.  I did have a won’t run on simulator error that I had to solve after switching back to stable and upgrading, but it appears all is good now.

I hate being the one causing unnecessary work, thank you for your quick response!
